posydon/binary_evol/SN/step_SN.py

```diff
@@ -1623,21 +1626,46 @@ def SNCheck(
                                   verbose=self.verbose)
 
             # update the binary object which was bound at least before the SN
+            #Check if this is the first SN
+            first_SN = binary.true_anomaly_first_SN is None
```

Comment:
A better way to check if this is the first or second supernova would be to check the stellar.state of the companion star. If the companion is already a compact object, i.e. `star.state in ['massless_remnant', 'WD', 'NS', 'BH']`, then this is the second supernova.

Reply:
The problem with this is that we use the term `massless_remnant` when referring to both a binary that has been disrupted as well as an initially single star. The ambiguity will need to be resolved in the future. For now though, let's use a different method. @chase-kimball, can you add two flags to the binary object that says whether the star has gone through SN1 and SN2? It's verbose but clear.

Reply:
I've added a flag binary.first_SN_already_occurred to binarystar.py